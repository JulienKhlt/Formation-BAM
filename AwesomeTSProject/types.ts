export interface cocktail {
  idDrink: string;
  strDrink: string;
  strTags: string | null;
  strCategory: string;
  strIBA: string;
  strAlcoholic: string;
  strGlass: string;
  strInstructions: string;
}
