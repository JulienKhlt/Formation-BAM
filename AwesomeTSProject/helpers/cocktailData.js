export const cocktails = [
  {
    idDrink: '11007',
    strDrink: 'Margarita',
    strTags: 'IBA,ContemporaryClassic',
    strCategory: 'Ordinary Drink',
    strIBA: 'Contemporary Classics',
    strAlcoholic: 'Alcoholic',
    strGlass: 'Cocktail glass',
    strInstructions:
      'Rub the rim of the glass with the lime slice to make the salt stick to it. Take care to moisten only the outer rim and sprinkle the salt on it. The salt should present to the lips of the imbiber and never mix into the cocktail. Shake the other ingredients with ice, then carefully pour into the glass.',
  },
  {
    idDrink: '11000',
    strDrink: 'Mojito',
    strTags:
      'IBA,ContemporaryClassic,Alcoholic,USA,Asia,Vegan,Citrus,Brunch,Hangover,Mild',
    strCategory: 'Cocktail',
    strIBA: 'Contemporary Classics',
    strAlcoholic: 'Alcoholic',
    strGlass: 'Highball glass',
    strInstructions:
      'Muddle mint leaves with sugar and lime juice. Add a splash of soda water and fill the glass with cracked ice. Pour the rum and top with soda water. Garnish and serve with straw.',
  },
];
