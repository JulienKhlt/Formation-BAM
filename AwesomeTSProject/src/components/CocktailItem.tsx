import React from 'react';
import {StyleSheet, View, Text, Image} from 'react-native';
import {cocktail} from '../../types';

interface cocktail_container {
  cocktail: cocktail;
}

export const CocktailItem = (props: cocktail_container) => {
  const cocktail: cocktail = props.cocktail;
  return (
    <View style={styles.main_container}>
      <Image style={styles.image} source={{uri: 'image'}} />
      <View style={styles.content_container}>
        <View style={styles.header_container}>
          <Text style={styles.title_text}>{cocktail.strDrink}</Text>
        </View>
        <View style={styles.description_container}>
          <Text style={styles.description_text} numberOfLines={6}>
            {cocktail.strInstructions}
          </Text>
        </View>
        <View style={styles.tag_container}>
          <Text style={styles.tag_text} numberOfLines={1}>
            {cocktail.strTags.replace(/,/g, ', ')}
          </Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  main_container: {
    height: 190,
    flexDirection: 'row',
  },
  image: {
    width: 140,
    height: 180,
    margin: 5,
    backgroundColor: 'gray',
  },
  content_container: {
    flex: 1,
    margin: 5,
  },
  header_container: {
    flex: 3,
    flexDirection: 'row',
  },
  title_text: {
    fontWeight: 'bold',
    fontSize: 20,
    flex: 1,
    flexWrap: 'wrap',
    paddingRight: 5,
  },
  description_container: {
    flex: 10,
    alignItems: 'center',
  },
  description_text: {
    fontStyle: 'italic',
    color: '#666666',
  },
  tag_container: {
    flex: 3,
  },
  tag_text: {
    textAlign: 'right',
    fontSize: 14,
  },
});
