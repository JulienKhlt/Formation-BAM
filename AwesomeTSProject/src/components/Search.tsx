import React from 'react';
import {View, TextInput, Button} from 'react-native';

export const Search = () => {
  return (
    <View>
      <TextInput placeholder="Titre du cocktail" />
      <Button title="Rechercher" onPress={() => {}} />
    </View>
  );
};
