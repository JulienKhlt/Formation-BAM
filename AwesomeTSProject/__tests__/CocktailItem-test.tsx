import React from 'react';
import renderer from 'react-test-renderer';
import {cocktails} from '../helpers/cocktailData';
import {CocktailItem} from '../src/components/CocktailItem';
import {cocktail} from '../types';

test('mojito renders correctly', () => {
  const mojito: cocktail = cocktails[1];
  const cocktailitem = renderer
    .create(<CocktailItem cocktail={mojito} />)
    .toJSON();
  expect(cocktailitem).toMatchSnapshot();
});
