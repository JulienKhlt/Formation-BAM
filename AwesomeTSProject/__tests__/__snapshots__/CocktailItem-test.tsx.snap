// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mojito renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "height": 190,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "image",
      }
    }
    style={
      Object {
        "backgroundColor": "gray",
        "height": 180,
        "margin": 5,
        "width": 140,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 3,
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "flex": 1,
            "flexWrap": "wrap",
            "fontSize": 20,
            "fontWeight": "bold",
            "paddingRight": 5,
          }
        }
      >
        Mojito
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 10,
        }
      }
    >
      <Text
        numberOfLines={6}
        style={
          Object {
            "color": "#666666",
            "fontStyle": "italic",
          }
        }
      >
        Muddle mint leaves with sugar and lime juice. Add a splash of soda water and fill the glass with cracked ice. Pour the rum and top with soda water. Garnish and serve with straw.
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 3,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "fontSize": 14,
            "textAlign": "right",
          }
        }
      >
        IBA, ContemporaryClassic, Alcoholic, USA, Asia, Vegan, Citrus, Brunch, Hangover, Mild
      </Text>
    </View>
  </View>
</View>
`;
